cmd/soci-snapshotter-grpc/main.go

```diff
@@ -137,8 +138,10 @@ func main() {
 		log.G(ctx).Warn("skipped snapshotter is supported check")
 	}
 
+	serverOpts := []grpc.ServerOption{grpc.ChainUnaryInterceptor(unaryNamespaceInterceptor)}
```

Comment:
Can we add a streaming interceptor as well? I don't think we're using it here, but I'm not sure if the containerd stream API would connect to SOCI via a stream as well. If we do add any streaming APIs, it would be nice to not have to go through this debugging again to rediscover the issue.

Reply:
Sure, ended up just moving all this to a new file so it doesn't clutter `main.go` too much. Can reverse this if needed.